### PR TITLE
Update coupled to latest UFS version

### LINKED ISSUE #230: Enable different coupled options (NSST on/off)
In order to turn the NSST models, including the diurnal warming and skin-layer cooling model, on in the coupled model, some modifications of the global-workflow are necessary, since the NSST model hasn't been tested systematically in the coupled frame. 

In principle, the NSST model is ready to run since it has been built in fv3atm and it is a component of the operational GFS (uncoupled). There is a switch to turn on the NSST model through a name list array. 

However, there is a fundamental issue to deal with when turning on the NSST model in the coupled system, i.e., how to combine the NSST T-Profile, particularly the diurnal warming profile,  with the oceanic model layer T-Profile to determine the foundation temperature, and then obtain the required underlying surface temperatures for the air-sea interface fluxes calculation for the coupled model and a radiative transfer model such as CRTM for the data assimilation.  
There is a scheme to determine the foundation temperature and the required surface temperatures, referred to Scheme-A, already built in the current fv3atm, specifically in the ccpp/physics submodule. 
The goal of this task is to make sure the coupled system work well the the NSST model  turned on, including the necessary modifications in the workflow, the test of the Scheme-A and the development of a more effective scheme, Scheme-B, to incorporate the NSST T-Profile into oceanic model layer T-Profile to provide a more realistic surface temperatures in fluxes calculation and the radiance simulation.

### LINKED ISSUE #324: update feature/coupled-crow to use latest ufs-weather-model
To update feature/coupled-crow to use the latest develop branch of ufs-weather-model, we need to track the changes in the tests/parm folder in the diffs here: 
https://github.com/ufs-community/ufs-weather-model/compare/Prototype-6.0beta...develop

Updates are needed to: 
* change name of mediator in nems.configure 
* MOM_input has new option ALLOW_LANDMASK_CHANGES (we use the default value which is false, but still should keep in sync)
* ice_in has changes
* the build needs to be updated as well (see https://github.com/JessicaMeixner-NOAA/global-workflow/commit/cdc04db78d946d904f4e1235956e48864c9a3ba3 for needed change)
* use ocean mesh file (a new ocean mesh fix file is not needed as its the same as the ice mesh file)

### LINKED ISSUE #325: update wave inputs
After the ufs-weather-model PR: https://github.com/ufs-community/ufs-weather-model/pull/573 is merged the WW3 input files will need to be updated for ww3_multi.inp (this is for develop branch and feature/coupled-crow if either want to run w/waves w/the this version of ufs-weather-model)

### LINKED ISSUE #333: coupled: updates to ice_in
Update the ice_in according the the PR in ufs-weather-model https://github.com/ufs-community/ufs-weather-model/pull/610 when updating ufs-weather-model to this commit.

### LINKED ISSUE #336: Update feature/coupled-crow to use the latest coupled aerosols app
Branch `feature/coupled-crow` needs to be updated to run the latest ufs-weather-model app including prognostic aerosols.

### LINKED ISSUE #337: Improperly set NST_SPINUP in FV3 namelist (coupled-crow)
In branch `feature/coupled-crow`, the `NST_SPINUP` flag in the parsed FV3 namelist  is always `0` regardless of the value set in the case YAML file.

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
Updates the coupled workflow to use the current version of UFS. This requires multiple updates to the configuration to handle new or changed settings.

Closes: #230, #324, #325, #333, #336, #337

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
Currently on hold until ufs-community/ufs-weather-model/pull/664 is merged to fix a CICE issue, at which point the checkout hash will need to be updated.

File: sorc/build_ufs_coupled.sh
Comment (JessicaMeixner-NOAA) [inline sorc/build_ufs_coupled.sh]: based on the https://github.com/NOAA-EMC/WW3/wiki/Commit-Queue the if statement for if the the GOCART directory exists is not going to work by the time the update for the ice history is merged in.

File: ush/parsing_namelists_CICE.sh
Comment (JessicaMeixner-NOAA) [inline ush/parsing_namelists_CICE.sh:212]: @DeniseWorthen @LydiaStefanova-NOAA Were these fields we had specifically turned off in the workflow?  Should we keep these off in the workflow or turn them back on?

Comment (LydiaStefanova-NOAA) [reply ush/parsing_namelists_CICE.sh:212]: I can't answer this - I don't know what these settings do, nor what they should be. Noticed in the configuration spreadsheet that a number of them were changed from 'mdhxx' to 'x' between P3.1 and P4, but don't know why.

Comment (JessicaMeixner-NOAA) [reply ush/parsing_namelists_CICE.sh:212]: @LydiaStefanova-NOAA this is for which fields we want in the ice history output files.  'x' means it's not included and then 'mdhxx' means we will have output for this.   These were turned off as you mentioned between P3.1 and P4 for the workflow, but have remained in the regression tests as its helpful for debugging.

Comment (WalterKolczynski-NOAA) [reply ush/parsing_namelists_CICE.sh:212]: There were a number of these types of changes. At first I just changed the ones that seem to have been updated since the last commit, then I noticed other settings also didn't match an naively updated them. We can revert 2dcf8f2b, which is the one where I made that second batch of changes (and then re-do the one mispaste of the new sw_redist setting that was also fixed there). Take a look at the changes from just that commit and let me know.

File: sorc/build_ufs_coupled.sh
Comment (WalterKolczynski-NOAA) [reply sorc/build_ufs_coupled.sh]: I'll change this back to using the -a switch. Unless after ufs-community/ufs-weather-model/pull/591 there will be a DAPP option that incorporates everything? That would be ideal.

Comment (JessicaMeixner-NOAA) [reply sorc/build_ufs_coupled.sh]: Looks to me like: ATMAERO was added only.  Seems like a good APP/feature to maybe request for the future?

File: ush/parsing_namelists_CICE.sh
Comment (JessicaMeixner-NOAA) [reply ush/parsing_namelists_CICE.sh:212]: @WalterKolczynski-NOAA it's a lot easier when these files just match up exactly.  I do want to say this saved us a fair amount of space on benchmark runs though.  @LydiaStefanova-NOAA I guess the question is, do we want the extra output for ice files in the full 168 runs, or do we want to keep the list that was trimmed down in P4 for ice?  Should we check with @DeniseWorthen Sulagna and @rgrumbine and others looking at ice output to see if we can continue without the extra ice output or if it'd be good to add these back -- if for nothing else to have more consistency with the rt system.

Comment (LydiaStefanova-NOAA) [reply ush/parsing_namelists_CICE.sh:212]: I went through the ice.nml section to list what's different from P6 - see spreadsheet [here](https://docs.google.com/spreadsheets/d/1CpJ7U33wIA-uWlqFewgBNbQ8AtAw0727d9c49QMiHnM/edit?usp=sharing). Red=newly turned on; Blue=newly turned off.

I went by [https://cesmcice.readthedocs.io/en/latest/users_guide/ice_history.html](url) to tie the parameters to descriptions. A couple of those were not on the list (f_dsnow and f_fbot). 

Of the newly turned off fields (8), we definitely want f_mlt_onset and f_frz_onset back on. Maybe f_divu and f_shear too. 

For the newly turned on fields (20) - most of them relate to radiation/albedo, salt, snow, ocean currents, atm winds. They may be useful to someone for debugging/sanity checking - I will leave it to others to weigh on that.

File: ush/parsing_namelists_CICE.sh
Comment (JessicaMeixner-NOAA) [inline ush/parsing_namelists_CICE.sh]: let's revert any that have gone from "on" to "off" (mdhxx->x), but anything that was turned on (x->mdhxx) as on.  

So the variables to revert are: f_divu, f_shear, f_divdt* (lines 296-299), f_mld_onset, f_frz_onset

Comment (WalterKolczynski-NOAA) [reply ush/parsing_namelists_CICE.sh]: Done

File: ush/parsing_namelists_CICE.sh
Comment (WalterKolczynski-NOAA) [reply ush/parsing_namelists_CICE.sh:212]: Per later comment from Jessica, all those that were turned off have been turned back on.

**Comment by WalterKolczynski-NOAA** [comment]
coupled_free_forecast_wave, aerosol_firex_forecast, and aerosol_free_forecast all report as succeeded on Hera.

File: sorc/build_all.sh
Comment (JessicaMeixner-NOAA) [inline sorc/build_all.sh:17]: Does the readme need to be updated for this?

Comment (WalterKolczynski-NOAA) [reply sorc/build_all.sh:17]: Yes. I'll get on that.

Comment (WalterKolczynski-NOAA) [reply sorc/build_all.sh:17]: Updated

**Review by JessicaMeixner-NOAA — APPROVED** [review]
For the coupled side of things, this all looks fine.  Approving based on the fact that @WalterKolczynski-NOAA had a successful test on hera.

**Review by rmontuoro — APPROVED** [review]
Looks good.

**Pull request merged**
